model_analyzer/monitor/dcgm/dcgm_monitor.py

```diff
@@ -0,0 +1,128 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from model_analyzer.record.gpu_free_memory import GPUFreeMemory
+from model_analyzer.record.gpu_used_memory import GPUUsedMemory
+from model_analyzer.monitor.monitor import Monitor
+from model_analyzer.model_analyzer_exceptions import \
+    TritonModelAnalyzerException
+
+import model_analyzer.monitor.dcgm.dcgm_agent as dcgm_agent
+import model_analyzer.monitor.dcgm.dcgm_fields as dcgm_fields
+import model_analyzer.monitor.dcgm.dcgm_field_helpers as dcgm_field_helpers
+import model_analyzer.monitor.dcgm.dcgm_structs as structs
+
+from multiprocessing.pool import ThreadPool
+import time
+
+
+class DCGMMonitor(Monitor):
+    """
+    Use DCGM to monitor GPU metrics
+    """
+
+    # Mapping between the DCGM Fields and Model Analyzer Records
+    MODEL_ANALYZER_TO_DCGM_FIELD = {
+        GPUUsedMemory: dcgm_fields.DCGM_FI_DEV_FB_USED,
+        GPUFreeMemory: dcgm_fields.DCGM_FI_DEV_FB_FREE
+    }
+
+    def __init__(self, frequency, tags, dcgmPath=None):
+        """
+        Parameters
+        ----------
+        frequency : int
+            Sampling frequency for the metric
+        tags : list
+            List of Record types to monitor
+        dcgmPath : str (optional)
+            DCGM installation path
+        """
+
+        super().__init__(frequency, tags)
+        structs._dcgmInit(dcgmPath)
+        dcgm_agent.dcgmInit()
+
+        # Start DCGM in the embedded mode to use the shared library
+        self.dcgm_handle = dcgm_handle = dcgm_agent.dcgmStartEmbedded(
+            structs.DCGM_OPERATION_MODE_MANUAL)
+
+        # Create DCGM monitor group
+        self.group_id = dcgm_agent.dcgmGroupCreate(dcgm_handle,
+                                                   structs.DCGM_GROUP_EMPTY,
+                                                   "triton-monitor")
+        # Add the GPUs to the group
+        for gpu in self._gpus:
+            dcgm_agent.dcgmGroupAddDevice(dcgm_handle, self.group_id,
+                                          gpu.device_id())
+
+        frequency = int(self._frequency * 1000)
+        fields = []
+        for tag in tags:
+            if tag in self.MODEL_ANALYZER_TO_DCGM_FIELD:
+                dcgm_field = self.MODEL_ANALYZER_TO_DCGM_FIELD[tag]
+                fields.append(dcgm_field)
+            else:
+                raise TritonModelAnalyzerException(
+                    f'{tag} is not supported by Model Analyzer DCGM Monitor')
+
+        self.dcgm_field_group_id = dcgm_agent.dcgmFieldGroupCreate(
+            dcgm_handle, fields, 'triton-monitor')
+
+        self.group_watcher = dcgm_field_helpers.DcgmFieldGroupWatcher(
+            dcgm_handle, self.group_id, self.dcgm_field_group_id.value,
+            structs.DCGM_OPERATION_MODE_MANUAL, frequency, 3600, 0, 0)
+
+    def _monitoring_iteration(self):
+        self.group_watcher.GetMore()
+
+    def _collect_records(self):
```

Comment:
I think we're including a short docstring for private functions

Reply:
I added a description in the parent class. I will add it here too.